rust/sealevel/programs/build-programs.sh

```diff
@@ -0,0 +1,93 @@
+#!/bin/bash
+
+# The script:
+# - uses the desired version of the solana cli for building programs
+# - resets the solana cli to the desired version after building programs
+# - builds the programs, accepting an arg for what types of programs to build
+# - if no arg is given, builds all programs
+
+# The first argument is the type of program to build
+PROGRAM_TYPE="${1:-all}"
+
+SOLANA_CLI_VERSION_FOR_BUILDING_PROGRAMS="1.14.20"
+
+# The paths to the programs
+CORE_PROGRAM_PATHS=("mailbox" "ism/multisig-ism-message-id" "validator-announce" "hyperlane-sealevel-igp")
+TOKEN_PROGRAM_PATHS=("hyperlane-sealevel-token" "hyperlane-sealevel-token-collateral" "hyperlane-sealevel-token-native")
+
+build_program () {
+    PROGRAM_PATH=$1
+    PREV_DIR=$(pwd)
+    log "Building $PROGRAM_PATH"
+    cd $PROGRAM_PATH
+    cargo build-sbf
+    cd $PREV_DIR
```

Comment:
i like to use pushd/popd for this type of thing - should be available on `bash` but not on the basic `sh`
```suggestion
    PROGRAM_PATH=$1
    log "Building $PROGRAM_PATH"
    pushd $PROGRAM_PATH
    cargo build-sbf
    popd
```